File: clase/clase_notificaciones.py
class Notificaciones:
    def __init__(self, id, usuario_id, mensaje, fecha_creacion, visto, usuario_nombre=None, usuario_foto=None):
        self.id = id
        self.usuario_id = usuario_id
        self.mensaje = mensaje
        self.fecha_creacion = fecha_creacion
        self.visto = visto
        self.usuario_nombre = usuario_nombre
        self.usuario_foto = usuario_foto

    def serialize(self):
        return {
            'id': self.id,
            'usuario_id': self.usuario_id,
            'mensaje': self.mensaje,
            'fecha_creacion': self.fecha_creacion,
            'visto': self.visto,
            'usuario_nombre': self.usuario_nombre,
            'usuario_foto': self.usuario_foto
        }

File: clase/clase_notificaciones.php
<?php
class Notificaciones {
    private $id;
    private $usuario_id;
    private $mensaje;
    private $fecha_creacion;
    private $visto;
    private $usuario_nombre;
    private $usuario_foto;

    public function __construct($data) {
        $this->id = $data['id'] ?? null;
        $this->usuario_id = $data['usuario_id'] ?? null;
        $this->mensaje = $data['mensaje'] ?? null;
        $this->fecha_creacion = $data['fecha_creacion'] ?? null;
        $this->visto = $data['visto'] ?? 0;
        $this->usuario_nombre = ($data['nombre'] ?? '') . ' ' . ($data['apellido'] ?? '');
        $this->usuario_foto = $data['foto'] ?? 'default-user.jpg';
    }

    public function serializar() {
        return [
            'id' => $this->id,
            'usuario_id' => $this->usuario_id,
            'mensaje' => $this->mensaje,
            'fecha_creacion' => $this->fecha_creacion,
            'visto' => $this->visto,
            'usuario_nombre' => $this->usuario_nombre,
            'usuario_foto' => $this->usuario_foto
        ];
    }
}